State/StatePatternClassDiagram.puml

```diff
@@ -0,0 +1,40 @@
+@startuml
+'https://plantuml.com/class-diagram
+
+interface Context {
+    + action1(): void
+    + action2(): void
+}
+
+class ConcreteContext implements Context {
+    - state: State
+
+    + action1(): void
+    + action2(): void
+
+}
+note left of ConcreteContext
+    action1 calls state.method1(),
+    action2 cals state.method2()
+end note
+
+
+interface State {
+    + method1(): void
+    + method2(): void
+}
+
+class ConcreteState1 implements State {
+
+    + ConcreteState1(): void
+    + method1(): void
+    + method2(): void
+}
```

Comment:
Changes LGTM, but this pattern has some cons that onestate implementation knows about the other state whereas strategy pattern doesn't know about the next set of state but it doesn't satisfies the this scenario. Though, It is breaking SRP and interface segregation principle but other way cause more harm when we have more switch/if condition which seems ok in this pattern. We can also think how best the state pattern can be implemented.

Reply:
state differs from strategy in many ways. Strategy is responsible for only a particular functinality such as SSO, SAML, password..for a same functionality. But in case of State. It refers to the position of a class by which state is assigned. State is directly conected to Context class every functionality